pkg/inventory/model/journal.go

```diff
@@ -0,0 +1,293 @@
+package model
+
+import (
+	liberr "github.com/konveyor/controller/pkg/error"
+	"github.com/konveyor/controller/pkg/ref"
+	"reflect"
+	"sync"
+)
+
+//
+// Event Actions.
+var (
+	Created int8 = 0x01
+	Updated int8 = 0x02
+	Deleted int8 = 0x04
+)
+
+//
+// Model event.
+type Event struct {
+	// The event subject.
+	Model Model
+	// The event action (created|updated|deleted).
+	Action int8
+}
+
+//
+// Event handler.
+type EventHandler interface {
+	// A model has been created.
+	Created(Model)
+	// A model has been updated.
+	Updated(Model)
+	// A model has been deleted.
+	Deleted(Model)
+	// An error has occurred delivering an event.
+	Error(error)
+	// An event watch has ended.
+	End()
+}
+
+//
+// Model event watch.
+type Watch struct {
+	// Model to be watched.
+	Model Model
+	// Event handler.
+	Handler EventHandler
+	// Event queue.
+	queue chan *Event
+	// Started
+	started bool
+}
+
+//
+// Match by model `kind`.
+func (w *Watch) Match(model Model) bool {
+	return ref.ToKind(w.Model) == ref.ToKind(model)
+}
+
+//
+// Queue event.
+func (w *Watch) notify(event *Event) {
+	if !w.Match(event.Model) {
+		return
+	}
+	defer func() {
```

Comment:
I'm not sure I understand the purpose of the closure around `recover()` here, could you explain it to me?

Reply:
When the watch is ended, the channel (queue) is closed.  A write on the closed channel will produce a panic which is dealt with by the recover().

Reply:
That makes sense, but what does the func() {} wrapping recover() do?

Reply:
So it can be deferred.